cmd/tpm-manager/tpm-manager.go

```
@@ -0,0 +1,507 @@
+	err = tpmhandler.Setup(&config)
+	if err != nil {
+		fmt.Printf("Unable to set up TPM handler: %v", err)
+		os.Exit(1)
+	}
+	configmap, err := client.ConfigMaps("kube-system").Get(ConfigName)
+	if err == nil && configmap != nil {
+		updateConfig(configmap)
+	}
+
+	addFlags(pflag.CommandLine)
+	leaderelection.BindFlags(&manager.leaderelect, pflag.CommandLine)
+	flag.InitFlags()
+
+	manager.client = client
+	manager.tpmhandler = tpmhandler
+	manager.recurringChan = make(chan int)
+
+	if !manager.leaderelect.LeaderElect {
+		run(wait.NeverStop)
+	}
+
+	id, err := os.Hostname()
+	if err != nil {
+		fmt.Printf("Unable to obtain hostname: %v", err)
+		return
+	}
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(client.Events(""))
+	recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "tpm-manager"})
+
+	leaderelection.RunOrDie(leaderelection.LeaderElectionConfig{
+		EndpointsMeta: api.ObjectMeta{
+			Namespace: "default",
+			Name:      "tpm-manager",
+		},
+		Client:        client,
+		Identity:      id,
+		EventRecorder: recorder,
+		LeaseDuration: manager.leaderelect.LeaseDuration.Duration,
+		RenewDeadline: manager.leaderelect.RenewDeadline.Duration,
+		RetryPeriod:   manager.leaderelect.RetryPeriod.Duration,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: run,
+			OnStoppedLeading: func() {
+				glog.Fatalf("leaderelection lost")
+			},
+		},
+	})
+}
+
+func nodeAddFn(obj interface{}) {
+	node, ok := obj.(*api.Node)
+	if !ok {
+		glog.Errorf("Node add got a non-Node")
+		return
+	}
+	verifyAndUpdate(node)
+}
+
+func configAddFn(obj interface{}) {
+	configmap, ok := obj.(*api.ConfigMap)
+	if !ok {
+		glog.Errorf("Config add got a non-ConfigMap")
+		return
+	}
+	if configmap.Name != ConfigName {
+		return
+	}
+	updateConfig(configmap)
+}
+
+func configUpdateFn(oldObj, newObj interface{}) {
+	configAddFn(newObj)
+}
+
+func updatePolicy() {
+	loadPolicy()
```

ignoring errors from this function